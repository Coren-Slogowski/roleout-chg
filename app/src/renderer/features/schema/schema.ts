import {DataAccess} from '../access/dataAccess'
import {SchemaOptions} from 'roleout-lib/build/objects/schema'
import {EnvironmentOptions} from '../options/options'

export interface Schema {
  readonly name: string
  readonly access: { [role: string]: DataAccess[] }
  readonly options: SchemaOptions
  readonly environmentOptions: EnvironmentOptions<SchemaOptions>
}