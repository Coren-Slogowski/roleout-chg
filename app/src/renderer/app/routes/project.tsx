import React, {FunctionComponent, useEffect} from 'react'
import {Outlet, useNavigate} from 'react-router-dom'
import ProjectLayout from '../../features/project/projectLayout'
import {useAppSelector} from '../hooks'

const Project: FunctionComponent<Record<string, unknown>> = () => {
  const navigate = useNavigate()

  const isLoaded = useAppSelector(state => state.project.isLoaded)

  useEffect(() => {
    if(!isLoaded) navigate('/')
  }, [isLoaded])

  if(!isLoaded) return <></>

  return (
    <ProjectLayout>
      <Outlet/>
    </ProjectLayout>
  )
}

export default Project
