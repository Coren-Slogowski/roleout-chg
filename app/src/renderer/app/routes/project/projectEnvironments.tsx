import React, {FunctionComponent} from 'react'
import {addEnvironment, removeEnvironment, updateEnvironment} from '../../../features/environments/environmentsSlice'
import EditableCollection from '../../../features/editableCollection/editableCollection'
import EnvironmentIcon from '@mui/icons-material/Widgets'

const ProjectEnvironments: FunctionComponent<Record<string, unknown>> = () => {
  return <EditableCollection itemType="Environment" Icon={EnvironmentIcon}
    selectorFn={(state) => state.environments} addAction={addEnvironment}
    removeAction={removeEnvironment} updateAction={updateEnvironment}/>
}

export default ProjectEnvironments
