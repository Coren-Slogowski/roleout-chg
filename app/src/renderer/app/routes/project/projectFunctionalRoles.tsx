import React, {FunctionComponent} from 'react'
import {
  addFunctionalRole,
  removeFunctionalRole,
  updateFunctionalRole
} from '../../../features/functionalRoles/functionalRolesSlice'
import EditableCollection from '../../../features/editableCollection/editableCollection'
import GroupIcon from '@mui/icons-material/PeopleAlt'

const ProjectFunctionalRoles: FunctionComponent<Record<string, unknown>> = () => {
  return <EditableCollection itemType="Functional Role" Icon={GroupIcon}
    selectorFn={(state) => state.functionalRoles} addAction={addFunctionalRole}
    removeAction={removeFunctionalRole} updateAction={updateFunctionalRole}/>
}

export default ProjectFunctionalRoles
