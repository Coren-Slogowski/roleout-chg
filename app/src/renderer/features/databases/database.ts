import {DatabaseOptions} from 'roleout-lib/build/objects/database'
import {Schema} from '../schema/schema'
import {EnvironmentOptions} from '../options/options'

export interface Database {
  readonly name: string
  readonly schemata: Schema[]
  readonly options: DatabaseOptions
  readonly environmentOptions: EnvironmentOptions<DatabaseOptions>
}