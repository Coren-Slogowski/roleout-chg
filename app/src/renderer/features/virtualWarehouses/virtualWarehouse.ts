import {VirtualWarehouseAccess} from '../access/virtualWarehouseAccess'
import {VirtualWarehouseOptions} from 'roleout-lib/build/objects/virtualWarehouse'

export interface VirtualWarehouse  {
  readonly name: string
  readonly options: VirtualWarehouseOptions
  readonly environmentOptions: {
    [env: string]: VirtualWarehouseOptions
  }
  readonly access: { [role: string]: VirtualWarehouseAccess[] }
}