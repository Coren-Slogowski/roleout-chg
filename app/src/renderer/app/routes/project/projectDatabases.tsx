import React, {FunctionComponent, useMemo, useState} from 'react'
import {Box, Button, Grid, List, ListItem, ListItemText, Paper} from '@mui/material'
import AddCircleIcon from '@mui/icons-material/AddCircle'
import {useAppDispatch, useAppSelector} from '../../hooks'
import DatabaseListItem from '../../../features/databases/databaseListItem'
import {addDatabase, selectDatabases} from '../../../features/databases/databasesSlice'
import Divider from '@mui/material/Divider'
import AdvancedOptionsSlider from '../../../features/options/advancedOptionsSlider'
import NewSnowflakeIdentifierTextField from '../../../features/forms/NewSnowflakeIdentifierTextField'

const ProjectDatabases: FunctionComponent<Record<string, unknown>> = () => {
  const databases = useAppSelector(selectDatabases)
  const environments = useAppSelector(state => state.environments)
  const enforceUnquotedIdentifiers = useAppSelector(state => state.project.enforceUnquotedIdentifiers)
  const dispatch = useAppDispatch()
  const [newDatabaseName, setNewDatabaseName] = useState('')

  const databasesContent = useMemo(() => {
    if (databases.length > 0) {
      return databases.map(db => <DatabaseListItem key={db.name} database={db}/>).flatMap(e => [<Divider
        key={e.key + '-divider'}/>, e]).slice(1)
    }
    return <ListItem><ListItemText secondary="Add a database above"/></ListItem>
  }, [databases])

  const handleNewDatabaseSubmit = () => {
    if(newDatabaseName === '') return
    dispatch(addDatabase({name: newDatabaseName, environments}))
    setNewDatabaseName('')
  }

  const handleNewDatabaseChange = (identifier: string) => {
    setNewDatabaseName(identifier)
  }

  return (
    <Box
      justifyContent="center"
      alignItems="center"
      margin="0 auto"
    >
      <Grid container gap={2} alignItems="center">
        <Grid item xs={6}>
          <NewSnowflakeIdentifierTextField value={newDatabaseName} onChange={handleNewDatabaseChange}
            onSubmit={handleNewDatabaseSubmit} enforceUnquotedIdentifiers={enforceUnquotedIdentifiers}
            label="Database Name" variant="filled" fullWidth={true}
          />
        </Grid>
        <Grid item xs={2}>
          <Button variant="contained" onClick={handleNewDatabaseSubmit} endIcon={<AddCircleIcon/>}>
            Add
          </Button>
        </Grid>
        <Grid item>
          <AdvancedOptionsSlider/>
        </Grid>
        <Grid item xs={12}>
          <Paper variant="outlined">
            <List>
              {databasesContent}
            </List>
          </Paper>
        </Grid>
      </Grid>
    </Box>
  )
}

export default ProjectDatabases
