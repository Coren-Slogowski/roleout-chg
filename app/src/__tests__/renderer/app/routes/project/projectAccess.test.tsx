import React from 'react'
import '@testing-library/jest-dom'
import {renderWithProviders} from '../../../../testUtils'
import ProjectAccess from '../../../../../renderer/app/routes/project/projectAccess'
import {defaultVirtualWarehouseOptions} from 'roleout-lib/build/objects/virtualWarehouse'
import {fireEvent, screen} from '@testing-library/react'
import {defaultDatabaseOptions} from 'roleout-lib/build/objects/database'
import {defaultSchemaOptions} from 'roleout-lib/build/objects/schema'
import {defaultProjectOptions} from 'roleout-lib/build/project'

describe('ProjectAccess', () => {
  it('should render correctly', async () => {
    const testDatabases = [
      {
        name: 'MY_DB',
        schemata: [
          {
            name: 'MY_SCHEMA',
            access: {},
            options: defaultSchemaOptions,
            environmentOptions: {}
          }
        ],
        options: defaultDatabaseOptions,
        environmentOptions: {}
      }
    ]
    const testEnvironments = ['PROD', 'DEV', 'TEST']
    const testRoles = ['ANALYST', 'ELT', 'DEVOPS']
    const testWarehouses = ['ANALYSIS', 'ELT', 'DEVOPS']

    const preloadedState = {
      databases: testDatabases,
      environments: testEnvironments.map(name => ({name})),
      functionalRoles: testRoles.map(name => ({name})),
      virtualWarehouses: testWarehouses.map(name => {
        return {
          name,
          access: {},
          options: defaultVirtualWarehouseOptions,
          environmentOptions: {}
        }
      }),
      project: {
        name: 'Test Project',
        isLoaded: true,
        environmentsEnabled: true,
        schemaObjectGroupsEnabled: true,
        enforceUnquotedIdentifiers: defaultProjectOptions.enforceUnquotedIdentifiers
      }
    }

    expect(renderWithProviders(<ProjectAccess/>, {preloadedState})).toMatchSnapshot()
    testDatabases.forEach(db => {
      expect(screen.getByText(db.name)).toBeInTheDocument()
      db.schemata.forEach(schema =>
        expect(screen.getByText(schema.name)).toBeInTheDocument()
      )
    }
    )
    testRoles.forEach(name => expect(screen.getByText(name)).toBeInTheDocument())
    expect(screen.getByText('PROD')).toBeInTheDocument()

    fireEvent.click(screen.getByText('Virtual Warehouses'))
    await screen.findAllByText('Virtual Warehouse')
    testWarehouses.forEach(name => expect(screen.getAllByText(name).length).toBeGreaterThan(0))
    testRoles.forEach(name => expect(screen.getAllByText(name).length).toBeGreaterThan(0))
  })
})