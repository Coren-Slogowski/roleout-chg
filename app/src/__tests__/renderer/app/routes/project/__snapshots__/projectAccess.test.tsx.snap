// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectAccess should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-wk7lf8"
      >
        <div
          class="MuiBox-root css-l256j9"
        >
          <div
            class="MuiFormControl-root css-tpwpig-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="env-select-label"
            >
              Environment
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="env-select-label env-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="env-select"
                role="button"
                tabindex="0"
              >
                PROD
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="PROD"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1in441m"
                >
                  <span>
                    Environment
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1gsv261"
        >
          <div
            class="MuiTabs-root css-1ujnqem-MuiTabs-root"
          >
            <div
              class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
              style="overflow: hidden; margin-bottom: 0px;"
            >
              <div
                aria-label="access tabs"
                class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-selected="true"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                  role="tab"
                  tabindex="0"
                  type="button"
                >
                  Databases
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  Virtual Warehouses
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
        </div>
        <div
          aria-labelledby="simple-tab-0"
          id="simple-tabpanel-0"
          role="tabpanel"
        >
          <div
            class="MuiTableContainer-root css-xoigi4-MuiTableContainer-root"
          >
            <table
              class="MuiTable-root css-rqglhn-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium header css-1ohq27p-MuiTableCell-root"
                    scope="col"
                  >
                    <h3>
                      Database
                    </h3>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium header css-1ohq27p-MuiTableCell-root"
                    scope="col"
                  >
                    <h3>
                      Schema
                    </h3>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium rotated css-1gs0q6-MuiTableCell-root"
                    scope="col"
                  >
                    <div>
                      <span
                        class="nowrap"
                      >
                        ANALYST
                      </span>
                    </div>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium rotated css-llf68u-MuiTableCell-root"
                    scope="col"
                  >
                    <div>
                      <span
                        class="nowrap"
                      >
                        ELT
                      </span>
                    </div>
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium rotated css-2hblwg-MuiTableCell-root"
                    scope="col"
                  >
                    <div>
                      <span
                        class="nowrap"
                      >
                        DEVOPS
                      </span>
                    </div>
                  </th>
                </tr>
              </thead>
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableBody-root css-7087ke-MuiPaper-root-MuiTableBody-root"
                role="rowgroup"
              >
                <tr
                  class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                  >
                    MY_DB
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                  />
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                  />
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                  />
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                  />
                </tr>
                <tr
                  class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                    rowspan="1"
                  />
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                    colspan="1"
                  >
                    MY_SCHEMA
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-14rtyod-MuiTableCell-root"
                  >
                     
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-14rtyod-MuiTableCell-root"
                  >
                     
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-14rtyod-MuiTableCell-root"
                  >
                     
                  </td>
                </tr>
              </div>
            </table>
          </div>
        </div>
        <div
          aria-labelledby="simple-tab-1"
          hidden=""
          id="simple-tabpanel-1"
          role="tabpanel"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-wk7lf8"
    >
      <div
        class="MuiBox-root css-l256j9"
      >
        <div
          class="MuiFormControl-root css-tpwpig-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            id="env-select-label"
          >
            Environment
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="env-select-label env-select"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="env-select"
              role="button"
              tabindex="0"
            >
              PROD
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="PROD"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1in441m"
              >
                <span>
                  Environment
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-1gsv261"
      >
        <div
          class="MuiTabs-root css-1ujnqem-MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            style="overflow: hidden; margin-bottom: 0px;"
          >
            <div
              aria-label="access tabs"
              class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="0"
                type="button"
              >
                Databases
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Virtual Warehouses
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </div>
      <div
        aria-labelledby="simple-tab-0"
        id="simple-tabpanel-0"
        role="tabpanel"
      >
        <div
          class="MuiTableContainer-root css-xoigi4-MuiTableContainer-root"
        >
          <table
            class="MuiTable-root css-rqglhn-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium header css-1ohq27p-MuiTableCell-root"
                  scope="col"
                >
                  <h3>
                    Database
                  </h3>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium header css-1ohq27p-MuiTableCell-root"
                  scope="col"
                >
                  <h3>
                    Schema
                  </h3>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium rotated css-1gs0q6-MuiTableCell-root"
                  scope="col"
                >
                  <div>
                    <span
                      class="nowrap"
                    >
                      ANALYST
                    </span>
                  </div>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium rotated css-llf68u-MuiTableCell-root"
                  scope="col"
                >
                  <div>
                    <span
                      class="nowrap"
                    >
                      ELT
                    </span>
                  </div>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium rotated css-2hblwg-MuiTableCell-root"
                  scope="col"
                >
                  <div>
                    <span
                      class="nowrap"
                    >
                      DEVOPS
                    </span>
                  </div>
                </th>
              </tr>
            </thead>
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableBody-root css-7087ke-MuiPaper-root-MuiTableBody-root"
              role="rowgroup"
            >
              <tr
                class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                >
                  MY_DB
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                />
              </tr>
              <tr
                class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                  rowspan="1"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-c5ket7-MuiTableCell-root"
                  colspan="1"
                >
                  MY_SCHEMA
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-14rtyod-MuiTableCell-root"
                >
                   
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-14rtyod-MuiTableCell-root"
                >
                   
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-14rtyod-MuiTableCell-root"
                >
                   
                </td>
              </tr>
            </div>
          </table>
        </div>
      </div>
      <div
        aria-labelledby="simple-tab-1"
        hidden=""
        id="simple-tabpanel-1"
        role="tabpanel"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
